addons/website_sale_product_configurator/controllers/main.py

```diff
@@ -1,101 +1,69 @@
+                'already_configured': already_configured,
+                'mode': kw.get('mode', 'add'),
+                'product_custom_attribute_values': kw.get('product_custom_attribute_values', None),
+                'no_attribute': kw.get('no_attribute', False),
+                'custom_attribute': kw.get('custom_attribute', False),
             }
-        """
-        if lang:
-            request.website = request.website.with_context(lang=lang)
-
-        order = request.website.sale_get_order(force_create=True)
-        if order.state != 'draft':
-            request.session['sale_order_id'] = None
-            order = request.website.sale_get_order(force_create=True)
+        )
 
-        product_and_options = json.loads(product_and_options)
-        if product_and_options:
-            # The main product is the first, optional products are the rest
-            main_product = product_and_options[0]
-            value = order._cart_update(
-                product_id=main_product['product_id'],
-                add_qty=main_product['quantity'],
-                product_custom_attribute_values=main_product['product_custom_attribute_values'],
-                no_variant_attribute_values=main_product['no_variant_attribute_values'],
-                **kwargs
-            )
-
-            if value['line_id']:
-                # Link option with its parent iff line has been created.
-                option_parent = {main_product['unique_id']: value['line_id']}
-                for option in product_and_options[1:]:
-                    parent_unique_id = option['parent_unique_id']
-                    option_value = order._cart_update(
-                        product_id=option['product_id'],
-                        set_qty=option['quantity'],
-                        linked_line_id=option_parent[parent_unique_id],
-                        product_custom_attribute_values=option['product_custom_attribute_values'],
-                        no_variant_attribute_values=option['no_variant_attribute_values'],
-                        **kwargs
-                    )
-                    option_parent[option['unique_id']] = option_value['line_id']
-
-        request.session['website_sale_cart_quantity'] = order.cart_quantity
+    @route(
+        '/sale_product_configurator/optional_product_items',
+        type='json', auth='public', methods=['POST'], website=True,
+    )
+    def optional_product_items(self, product_id, add_qty=1, **kw):
+        product = request.env['product.product'].browse(int(product_id))
 
-        return str(order.cart_quantity)
+        return request.env['ir.ui.view']._render_template(
+            'website_sale_product_configurator.optional_product_items',
+            {
+                'product': product,
+                'parent_name': product.name,
+                'parent_combination': product.product_template_attribute_value_ids,
+                'add_qty': float(add_qty) or 1.0,
```

Comment:
default value of kwarg
```suggestion
                'add_qty': float(add_qty),
```